File: 179-largest-number/179-largest-number.ts
function largestNumber(nums: number[]): string {
  const N = nums.length,
        l = console.log;
  
  const res = nums
    .map(String)
    .sort((a, b) => (a.concat(b) > b.concat(a)) ? -1 : 1)
    .join('');
  
  return res.at(0) == '0' ? '0' : res
};

File: 179-largest-number/179-largest-number.java
class Solution {
    public String largestNumber(int[] nums) {
        String[] strArr = new String[nums.length];
        
        for (int i = 0; i < nums.length; i++)
            strArr[i] = Integer.toString(nums[i]);

        // Arrays.sort(strArr, (a, b) -> (b + a).compareTo((a + b)));
        Arrays.sort(strArr, (a, b) -> b.concat(a).compareTo(a.concat(b)));
        String ans = String.join("", strArr);

        if (ans.charAt(0) == '0')
            ans = "0";

        return ans;
    }
}